casper/src/main/scala/coop/rchain/casper/blocks/BlockReceiver.scala

```diff
@@ -0,0 +1,280 @@
+package coop.rchain.casper.blocks
+
+import cats.effect.concurrent.Ref
+import cats.effect.{Concurrent, Sync}
+import cats.syntax.all._
+import coop.rchain.blockstorage.BlockStore
+import coop.rchain.blockstorage.BlockStore.BlockStore
+import coop.rchain.blockstorage.dag.BlockDagStorage
+import coop.rchain.casper.engine.BlockRetriever
+import coop.rchain.casper.protocol.BlockMessage
+import coop.rchain.casper.syntax._
+import coop.rchain.casper.util.ProtoUtil
+import coop.rchain.casper.{CasperShardConf, PrettyPrinter, Validate}
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.shared.Log
+import coop.rchain.shared.syntax._
+import fs2.Stream
+import fs2.concurrent.Queue
+
+sealed trait RecvStatus
+// Checking if block should be ignored or continue with processing
+case object BeginReceived extends RecvStatus
+// Block received and stored in blocks store, waiting for validation and DAG insertion
+case object ReceivedAndStored extends RecvStatus
+// Block sent to validation
+case object PendingValidation extends RecvStatus
+// Requested missing dependencies
+case object PendingRequest extends RecvStatus
```

Comment:
Why not just `Requested`?

Reply:
I'm trying to avoid collisions with existing names in the project. Maybe some prefix or suffix would be useful like `RequestedRecvStatus`.

Reply:
Why to do that? If this status is used outside it can be renamed in the calling code, isn't it?